guard.cpp

```diff
@@ -18,7 +18,8 @@ void guardList()
     std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
     for (const auto& elem : records)
     {
-        if (elem.errType != GARD_Reconfig)
+        if ((elem.errType != GARD_Reconfig) &&
```

Comment:
As per discussion changed the clearall option and marking all records as resolved. User can erase one by one resolved records. You want me to change this to earlier version i.e. to do a real clear of the guard records or add a new option to be used for lab purpose?

Reply:
As per discussion adding new option to erase the guard records.

Reply:
Clearall user expectation is to clear all guard record from memory . here , you are ot clearing instead , changing to resolved. Istead of changing existing behavior , better to use new option to change the state. This will avoid changes in the existing tool user and automation scripts.

Reply:
We are trying to make guard tool same as in FSP, as previously we were not invalidating the resources, we are correcting that behavior in this PR.
New guard options:

Usage: guard [OPTIONS]

Options:
  -h,--help                   Guard CLI tool options
  -c,--create TEXT            Create Guard record, expects physical path as input
  -i,--invalidate TEXT        Invalidate a single Guard record, expects physical path as input
  -I,--invalidate-all         Invalidates all the Guard records
  -l,--list                   List all the Guard'ed resources
  -r,--reset                  Erase all the Guard records
  -v,--version                Version of GUARD tool